modules/core/src/main/scala/lucuma/odb/api/model/TargetModel.scala

```diff
@@ -228,69 +246,124 @@ object TargetModel extends TargetOptics {
 
   }
 
-  final case class EditNonsidereal(
-    targetId:  Target.Id,
-    existence: Option[Existence],
-    name:      Option[NonEmptyString],
-    key:       Option[EphemerisKey],
+  /**
+   * Input required to edit either a non-sidereal or sidereal target.
+   */
+  final case class Edit(
+    nonsidereal: Option[EditNonsidereal],
+    sidereal:    Option[EditSidereal],
+    rename:      Option[EditName]
```

Comment:
Will this apply to either sideral/non-sidereal?

Reply:
Yes, `rename` will execute this state editor: `lucuma.core.model.Target.name.assign_(newName)` in the end.  `Target` name is the same for all targets regardless of their `track`ing.